disk_objectstore/backup_utils.py

```diff
@@ -0,0 +1,382 @@
+"""
+Utilities to back up a container.
+"""
+
+import datetime
+import logging
+import random
+import shutil
+import sqlite3
+import string
+import subprocess
+import tempfile
+from pathlib import Path
+from typing import Callable, Optional
+
+from disk_objectstore.container import Container
+
+logging.basicConfig(format="%(levelname)s:%(message)s")
+backup_logger = logging.getLogger(__name__)
+
+
+class BackupError(Exception):
+    "Raised when backup fails."
+
+
+def split_remote_and_path(dest: str):
+    """extract remote and path from <remote>:<path>"""
+    split_dest = dest.split(":")
+    if len(split_dest) == 1:
+        return None, Path(dest)
+    if len(split_dest) == 2:
+        return split_dest[0], Path(split_dest[1])
+    # more than 1 colon:
+    raise BackupError("Invalid destination format: <remote>:<path>")
+
+
+def is_exe_found(exe: str) -> bool:
+    return shutil.which(exe) is not None
+
+
+class BackupManager:
+    """
+    Class that contains all configuration and utility functions to create
+    backups except for the backup function itself, which is passed in according
+    to what is backed up (e.g. the disk-objectstore container in for this repo,
+    or the whole aiida storage in aiida-core)
+    """
+
+    def __init__(
+        self,
+        dest: str,
```

Comment:
It can contain the remote specifier (e.g. `remote:/path`) so string has to be supported. I think keeping it as str is cleaner as also the cli only takes string (although i have specified `type=click.Path()`, but maybe that should be changed)

Reply:
Ah yeah, fair enough. Leave it as is